### ISSUE TITLE
using step_on_plateau breaks training

### ISSUE THREAD

**Issue by my89** [allenai/allennlp#1226]
Hi,

Using a lr_scheduler that steps_on_plateau now breaks because of line 459 in allennlp/training/trainer.py . The code there makes sense for schedules based on the number of steps but not here (and passing None as the first argument breaks the plateau lr scheduler).
